service/health/response.go

```diff
@@ -14,26 +14,40 @@ type ClusterStatus struct {
 	State  key.LifecycleState `json:"state"`
 }
 
-// NodeStatus holds data about the health/status of a node.
-type NodeStatus struct {
-	// Identity
+// NodeStatusComputeResources holds data about available or requested compute resources.
+type NodeStatusComputeResources struct {
+	CPU         int64 `json:"cpu"`
+	MemoryBytes int64 `json:"memory_bytes"`
+}
+
+// NodeStatusMachineResources holds data about the status of storage on a cluster node.
+type NodeStatusMachineResources struct {
+	CPUCount               int64 `json:"cpu_count"`
+	MemoryCapacityBytes    int64 `json:"memory_capacity_bytes"`
+	MemoryAllocatableBytes int64 `json:"memory_allocatable_bytes"`
+	EphemeralStorageCap    int64 `json:"ephemeral_storage_capacity_bytes"`
+	EphemeralStorageAvail  int64 `json:"ephemeral_storage_allocatable_bytes"`
+}
+
+// NodeStatusIdentity holds data about the identity of a node which is generally static.
+type NodeStatusIdentity struct {
 	Name               string `json:"name"`
+	KubeletVersion     string `json:"kubelet_version"`
+	OperatorVersion    string `json:"operator_version"`
 	Role               string `json:"role"`
 	IP                 string `json:"ip"`
 	Hostname           string `json:"hostname"`
 	InstanceType       string `json:"instance_type"`
 	AvailabilityRegion string `json:"availability_region"`
 	AvailabilityZone   string `json:"availability_zone"`
-	// Versions
-	KubeletVersion  string `json:"kubelet_version"`
-	OperatorVersion string `json:"operator_version"`
-	// Resources
-	CPUCount               int64 `json:"cpu_count"`
-	MemoryCapacityBytes    int64 `json:"memory_capacity_bytes"`
-	MemoryAllocatableBytes int64 `json:"memory_allocatable_bytes"`
-	EphemeralStorageCap    int64 `json:"ephemeral_storage_capacity_bytes"`
-	EphemeralStorageAvail  int64 `json:"ephemeral_storage_allocatable_bytes"`
-	// Status
-	Ready  bool       `json:"ready"`
-	Health key.Health `json:"health"`
+}
+
+// NodeStatus represents the health status of a single node in a cluster.
+type NodeStatus struct {
```

Comment:
Can we put this type definition before member types definitions? That would be easier for me to read as this is what we did so far.